api-objects/Appointment.ts

```diff
@@ -34,4 +39,48 @@ export class Appointment {
     public ths_sectionc: number;
     public ths_sectionc_example: string;
     public ts_type: string;
+
+    public createTfiMap() {
+        return new Map<string, number>([['overallTFI', this.tfi_overallscore], ['intrusive', this.tfi_i], ['sense', this.tfi_sc], ['cognitive', this.tfi_c], ['sleep', this.tfi_si], ['auditory', this.tfi_a], ['relaxation', this.tfi_r], ['quality', this.tfi_q], ['emotional', this.tfi_e]]);
+    }
+
+    public createThsScoreMap() {
+        return new Map<string, number>([['thsA', this.ths_sectiona], ['thsB', this.ths_sectionb], ['thsC', this.ths_sectionb]]);
+    }
+    public createThsTextVar() {
+        return new Map<string, string>([['thsCtxt', this.getthsCtxt()], ['thsCex', this.ths_sectionc_example]]);
+    }
+
+    public testSeverityVars() {
+        return new Map<string, string>([['audiogramType', this.audiogramtype], ['leftHighSev', this.leftear_highf_severity], ['leftLowSev', this.leftear_lowf_severity], ['rightHighSev', this.rightear_highf_severity], ['rightLowSev', this.rightear_lowf_severity], ['otoscopyType', this.otoscopytype], ['tympanometryType', this.tympanometrytype]]);
+    }
+
+    public testConfigVars() {
+        return new Map<string, string>([['leftHighConfig', this.leftear_highf_configuration.replace(/;/g, ', ')], ['leftLowConfig', this.leftear_lowf_configuration.replace(/;/g, ', ')], ['rightHighConfig', this.rightear_highf_configuration.replace(/;/g, ', ')], ['rightLowConfig', this.rightear_lowf_configuration.replace(/;/g, ', ')]]);
+    }
+    answerStrings = new ThsAnswerStrings();
```

Comment:
what is this piece of code doing here? also why on earth is ThsAnswerStrings not a singleton class or non-constructed class?

Reply:
Lol, because midnight and I don't mix very well when it comes to good programming practice. Thanks for the catch!